addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -3980,6 +3994,89 @@ export class OdooEditor extends EventTarget {
         }
     }
 
+    /**
+     * Places the cursor in a safe place (not the editable root).
+     * Inserts an empty paragraph when there's no other way to insert text
+     * before/after a block.
+     * If selection was result of an arrow key press, no new 'P' is inserted.
+     *
+     * @param {Selection} selection
+     * @param {String} currentKeyPress
+     */
+    _fixSelectionOnEditableRoot(selection, currentKeyPress) {
+        if (!selection || selection.anchorNode !== this.editable) return;
+        const nodesToSkip = ['#text', "#comment"];
+        if (currentKeyPress && currentKeyPress.startsWith("Arrow")) {
+            nodesToSkip.push('HR');
+        }
+        let nodeAfterCursor = this.editable.childNodes[selection.anchorOffset];
+        let nodeBeforeCursor = nodeAfterCursor && nodeAfterCursor.previousSibling;
+        while (nodeAfterCursor && nodesToSkip.includes(nodeAfterCursor.nodeName)) {
+            nodeAfterCursor = nodeAfterCursor.nextSibling;
+        }
+        while (nodeBeforeCursor && nodesToSkip.includes(nodeBeforeCursor.nodeName)) {
+            nodeBeforeCursor = nodeBeforeCursor.previousSibling;
+        }
+        // Utils
+        const restoreLastSelection = () => {
+            const lastSelection = this._latestComputedSelectionInEditable;
+            if (!lastSelection || lastSelection.anchorNode === this.editable) {
+                selection.removeAllRanges();
+                return;
+            }
+            selection.setBaseAndExtent(lastSelection.anchorNode, lastSelection.anchorOffset,
+                lastSelection.focusNode, lastSelection.focusOffset);
+        }
```

Comment:
Wouldn't `historySetSelection` serve your purpose here wihout the need for a new local function ?